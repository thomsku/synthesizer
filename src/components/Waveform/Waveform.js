import React from "react";

class Waveform extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }
  render() {
    return (
      <div></div>
    );
  }
}

export default Waveform;
