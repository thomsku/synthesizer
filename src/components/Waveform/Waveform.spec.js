import React from "react";
import {shallow} from "enzyme";
import Waveform from "./Waveform";

describe("Waveform", () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Waveform setWaveform={jest.fn()} waveform="sine"/>));

  it("should render a <div />", () => {
    expect(wrapper.find("div")).toHaveLength(1);
  });
});
