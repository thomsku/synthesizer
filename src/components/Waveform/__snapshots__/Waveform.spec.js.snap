// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Waveform should render correctly 1`] = `
<div
  className="radiobuttons"
>
  <WithStyles(WithFormControlContext(ForwardRef(FormLabel)))
    className="parametertitle"
    component="legend"
  >
    Waveform
  </WithStyles(WithFormControlContext(ForwardRef(FormLabel)))>
  <ForwardRef(RadioGroup)
    aria-label="Waveform"
    className="radiogroup"
    name="waveform"
    onChange={[MockFunction]}
    value="sine"
  >
    <WithStyles(WithFormControlContext(ForwardRef(FormControlLabel)))
      className="radiolabel"
      control={
        <ForwardRef(WithStyles)
          className="radio"
        />
      }
      label="sine"
      value="sine"
    />
    <WithStyles(WithFormControlContext(ForwardRef(FormControlLabel)))
      className="radiolabel"
      control={
        <ForwardRef(WithStyles)
          className="radio"
        />
      }
      label="square"
      value="square"
    />
    <WithStyles(WithFormControlContext(ForwardRef(FormControlLabel)))
      className="radiolabel"
      control={
        <ForwardRef(WithStyles)
          className="radio"
        />
      }
      label="sawtooth"
      value="sawtooth"
    />
    <WithStyles(WithFormControlContext(ForwardRef(FormControlLabel)))
      className="radiolabel"
      control={
        <ForwardRef(WithStyles)
          className="radio"
        />
      }
      label="triangle"
      value="triangle"
    />
  </ForwardRef(RadioGroup)>
</div>
`;
